app/models/reports/repository.rb

```diff
@@ -120,21 +133,45 @@ def reporting_schema_v2?
       end
     end
 
+    private def registered_patients_query_v2(region)
+      Reports::FacilityState.for_facility(region).order(:month_date).pluck(:month_date, :monthly_registrations)
+    end
+
+    private def cumulative_registered_patients_query_v2(region)
+      Reports::FacilityState.for_facility(region).order(:month_date).pluck(:month_date, :cumulative_registrations)
+    end
+
     # Returns the full range of registered patient counts for a Region. We do this via one SQL query for each Region, because its
     # fast and easy via the underlying query.
     memoize def complete_monthly_registrations
-      items = regions.map { |region| RegionEntry.new(region, __method__, period_type: period_type) }
-      cache.fetch_multi(*items, force: bust_cache?) { |entry|
-        registered_patients_query.count(entry.region, period_type)
-      }
+      if reporting_schema_v2?
+        regions.each_with_object({}) { |region, result|
+          result[region.slug] = registered_patients_query_v2(region).each_with_object(Hash.new(0)) { |(month_date, count), hsh|
+            hsh[Period.month(month_date)] = count
+          }
```

Comment:
Is this worth extracting? Keying the results by `region.slug` and `month_date` would happen often enough (twice in this PR) right?